src/dfx/src/lib/builders/custom.rs

```diff
@@ -202,16 +201,10 @@ impl CanisterBuilder for CustomBuilder {
     }
 }
 
-fn run_command(args: Vec<String>, vars: &[super::Env<'_>], cwd: &Path) -> DfxResult<()> {
-    let (command_name, arguments) = args.split_first().unwrap();
-    let canonicalized = cwd
-        .join(command_name)
-        .canonicalize()
-        .or_else(|_| which::which(command_name))
-        .map_err(|_| anyhow!("Cannot find command or file {command_name}"))?;
-    let mut cmd = Command::new(&canonicalized);
-
-    cmd.args(arguments)
+fn run_shell_command(cmd_str: &str, vars: &[super::Env<'_>], cwd: &Path) -> DfxResult<()> {
+    let mut cmd = Command::new("sh");
+    cmd.arg("-c")
+        .arg(cmd_str)
```

Comment:
Some of the test failures are due to the fact that the previous code goes to some trouble to make sure that if you put `<name of executable in your project directory>` in your build script, dfx calls it, rather then reporting an error because you didn't put `./<name of executable in your project directory>`.

That's what `cwd.join(command_name).canonicalize()` is about.

Reply:
I see! that makes sense. I'll re-add something like that.